pages/deliverer/incoming-orders/[id].tsx

```diff
@@ -0,0 +1,201 @@
+        )
+        .eq('id', orderId)
+        .is('deliverer_id', null)
+        .limit(1)
+        .single()
+
+      if (ordersError) {
+        router.push('/home')
+        setAlert({
+          type: 'warning',
+          message: ordersError.code + ': ' + ordersError.message,
+        })
+        console.log(ordersError)
+        return
+      }
+
+      // means a deliverer has already takent the order
+      if (ordersData.deliverer_id) {
+        router.push('/home')
+        setAlert({ type: 'info', message: 'Order has been taken' })
+        return
+      }
+
+      console.log(ordersData)
+
+      setOrder({ ...ordersData, ordered_at: new Date(ordersData.ordered_at) })
+    }
+
+    query()
+
+    //not best practice, need to figure out how to handle this
+  }, [])
+
+  async function handleReject() {
+    console.log(
+      'store in db rejected, never show this order on the homescreen again',
+    )
+    let error = ''
+    const { error: removePendingError } = await supabaseClient.rpc(
+      'remove_pending_order',
+      { order_id: id, user_id: user.id },
+    )
+    if (removePendingError) {
+      error +=
+        removePendingError.code + ': ' + removePendingError.message + '\n'
+    }
+    const { error: addRejectedError } = await supabaseClient.rpc(
+      'add_rejected_user',
+      { order_id: id, user_id: user.id },
+    )
+    if (addRejectedError) {
+      error += addRejectedError.code + ': ' + addRejectedError.message
+    }
+
+    if (error) {
+      setAlert({ type: 'warning', message: error })
+    } else {
+      setAlert({ type: 'info', message: 'Rejected order' })
+    }
+    router.push('/home')
+  }
+
+  async function handleAccept() {
+    // implement auth to get the user uid
+    const { data: updatedOrdersData, error: updatedOrdersError } =
+      await supabaseClient
+        .from('orders')
+        .update({ id: id, deliverer_id: user.id })
+        .eq('id', id)
+        .is('is_active', true)
+        .is('deliverer_id', null)
+
+    // possible that order has been taken or completed
+    if (updatedOrdersError) {
+      const { data: ordersData, error: ordersError } = await supabaseClient
+        .from<IncomingOrderDto>('orders')
+        .select(
+          'id, restaurant_id, order_text, deliverer_id, ordered_at, is_active, destination_id, cost',
+        )
+        .filter('id', 'eq', orderId)
+        .filter('deliverer_id', 'is', 'NULL')
+      if (ordersError) {
+        router.push('/home')
+        setAlert({
+          type: 'warning',
+          message: ordersError.code + ': ' + ordersError.message,
+        })
+        console.log(ordersError)
+      }
+
+      if (ordersData.length == 0) {
+        router.push('/home')
+        setAlert({ type: 'info', message: 'Order has been taken' })
+      }
+    } else {
+      router.push('/home')
+      setAlert({ type: 'success', message: 'Accepted order' })
```

Comment:
we will actually need to update the order in the DB here right? TODO?

Reply:
```
const { data: updatedOrdersData, error: updatedOrdersError } =
      await supabaseClient
        .from('orders')
        .update({ id: id, deliverer_id: user.id })
        .eq('id', id)
        .is('is_active', true)
        .is('deliverer_id', null)
```
this part tries to update the database with the deliverer_id, then pushes to /home with a success message if there is no error